tests/dispense.rs

```diff
@@ -253,10 +277,76 @@ async fn many_concurrent_requests() {
         .into_iter()
         .filter(|tx| !matches!(tx.transaction, TransactionType::Mint(_)))
         .collect::<Vec<_>>();
+    assert_eq!(3, txs.len());
 
-    assert_eq!(COUNT, txs.len());
-    assert_eq!(
-        test_balance,
-        COUNT as u64 * context.faucet_config.dispense_amount
-    );
+    for recipient in recipient_addresses {
+        let test_balance = context
+            .provider
+            .get_asset_balance(&recipient.into(), context.faucet_config.dispense_asset_id)
+            .await
+            .unwrap();
+
+        assert_eq!(test_balance, context.faucet_config.dispense_amount);
+    }
+}
+
+#[tokio::test]
+async fn dispense_once_per_day() {
+    let mut rng = StdRng::seed_from_u64(42);
+    let recipient_address: Address = rng.gen();
+    let recipient_address_str = format!("{}", &recipient_address);
+    let context = TestContext::new(rng).await;
+    let addr = context.addr;
+
+    let dispense_interval = 24 * 60 * 60;
+    let time_increment = dispense_interval / 6;
+
+    let response = reqwest::Client::new()
+        .post(format!("http://{addr}/dispense"))
+        .json(&json!({
+            "captcha": "",
+            "address": recipient_address_str.clone(),
+        }))
+        .send()
+        .await
+        .expect("First dispensing request should be successful");
+
+    assert_eq!(response.status(), reqwest::StatusCode::CREATED);
+
+    for _ in 0..5 {
+        //tokio::time::advance(Duration::from_secs(time_increment)).await;
+        context.clock.advance(time_increment);
+
+        let response = reqwest::Client::new()
+            .post(format!("http://{addr}/dispense"))
+            .json(&json!({
+                "captcha": "",
+                "address": recipient_address_str.clone(),
+            }))
+            .send()
+            .await
+            .expect("Subsequent dispensing requests should be successful with status 500");
+
+        assert_eq!(
+            response.status(),
+            reqwest::StatusCode::INTERNAL_SERVER_ERROR
```

Comment:
Returning a 500 error seems a bit odd to me. Generally, 500 means the server encountered a runtime error and it means there is something broken on our end! I think something more appropriate would be a [Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429) error (status code 429). 4xx series errors indicate that the end user did something invalid - in this case, made requests beyond the limit.

Reply:
Agreed, it makes a lot more sense. Thanks!